tests/__init__.py

```diff
@@ -551,5 +552,106 @@ def test_batch_lock_key(self):
         Worker(self.tiger).run(once=True)
         self._ensure_queues(queued={'batch': 0})
 
+class TaskTestCase(BaseTestCase):
+    """
+    Task class test cases.
+    """
+    def test_delay(self):
+        task = Task(self.tiger, simple_task)
+        self._ensure_queues()
+        task.delay()
+        self._ensure_queues(queued={'default': 1})
+
+        # Canceling only works for scheduled tasks.
+        self.assertRaises(TaskNotFound, task.cancel)
+
+    def test_delay_scheduled(self):
+        task = Task(self.tiger, simple_task, queue='a')
+        task.delay(when=datetime.timedelta(minutes=5))
+        self._ensure_queues(scheduled={'a': 1})
+
+        # Test canceling a scheduled task.
+        task.cancel()
+        self._ensure_queues()
+
+        # Canceling again raises an error
+        self.assertRaises(TaskNotFound, task.cancel)
+
+    def test_delay_scheduled_2(self):
+        task = Task(self.tiger, simple_task, queue='a')
+        task.delay(when=datetime.timedelta(minutes=5))
+        self._ensure_queues(scheduled={'a': 1})
+
+        task_id = task.id
+
+        # We can't look up a non-unique task by recreating it.
+        task = Task(self.tiger, simple_task, queue='a')
+        self.assertRaises(TaskNotFound, task.cancel)
```

Comment:
The difference between this scenario and the one in the test is that this scenario actually commits a given ID to Redis. The previous scenario generates a random ID during initialization and then tries to retrieve it.

I guess it's not worth a hassle/code complexity to differentiate between all these scenarios just to avoid a quick Redis call (though I still think it could result in a more obvious exception).